unipacker/headers.py

```diff
@@ -477,10 +482,20 @@ def hdr_write(uc, base_addr, header, array_pos, **fields):
             setattr(header_struct, k, v)
 
 
-def pe_write(uc, base_addr, total_size, filename):
-    data = uc.mem_read(base_addr, total_size)
-    with open(filename, 'wb+') as f:
-        f.write(data)
+class pe_write:
```

Comment:
using a class instead (folllowing the way `open` works) so it works with/without context:

```python
with pe_write(...):
    pass

pe_write(...)


```